crates/scroll/primitives/src/l1_transaction.rs

```diff
@@ -0,0 +1,308 @@
+//! Scroll L1 message transaction
+
+use alloy_consensus::Transaction;
+use alloy_primitives::{
+    private::alloy_rlp::{Encodable, Header},
+    Address, Bytes, ChainId, PrimitiveSignature as Signature, TxKind, B256, U256,
+};
+use alloy_rlp::Decodable;
+use bytes::BufMut;
+use serde::{Deserialize, Serialize};
+#[cfg(any(test, feature = "reth-codec"))]
+use {reth_codecs::Compact, reth_codecs_derive::add_arbitrary_tests};
+
+/// L1 message transaction type id.
+pub const L1_MESSAGE_TRANSACTION_TYPE: u8 = 0x7E;
```

Comment:
Is there a reason we define this using hex? Happy to keep it as is but lets add to the doc comment that this refers to a decimal value of 126 for convenience.